packages/keystone/lib/adapters/index.js

```diff
@@ -21,6 +21,9 @@ class BaseKeystoneAdapter {
     // Connect to the database
     await this._connect({ name }, this.config);
 
+    // Validate the minimum database version requirements are met.
+    this.checkDatabaseVersion();
+
     // Set up all list adapters
     try {
       const taskResults = await this.postConnect();
```

Comment:
Ideally `this.checkDatabaseVersion()` is included inside the `try {` so the code for disconnecting from the database is always executed;

```javascript
try {
  // Validate the minimum database version requirements are met.
  await this.checkDatabaseVersion();
  const taskResults = await this.postConnect();
```

Reply:
Done.